tools/applicationinsights-web-snippet/src/miniLoader.ts

```diff
@@ -0,0 +1,401 @@
+            tags[strAiDevice + "id"] = type[strToLowerCase]();
+            tags[strAiDevice + "type"] = type;
+            tags[strAiOperationName] = locn && locn.pathname || "_unknown_";
+            tags[strAiSdkVersion] = "javascript:snippet_" + (appInsights.sv || appInsights.version);
+
+            let envelope:IEnvelope = {
+                time: _getTime(),
+                iKey: iKey,
+                name: "Microsoft.ApplicationInsights." + iKey.replace(/-/g, strEmpty) + "." + theType,
+                sampleRate: 100,
+                tags: tags,
+                data: {
+                    baseData: {
+                        ver: 2
+                    }
+                },
+                ver: 4.0,
+                seq: "1",
+                aiDataContract: undefined
+            };
+
+            return envelope;
+        }
+
+        function _createInternal(iKey:string, message:string, targetSrc:string, endpointUrl:any) {
+            let envelope : IEnvelope = _createEnvelope(iKey, "Message");
+            let data = envelope.data;
+            data.baseType = "MessageData";
+            let baseData = data.baseData;
+
+            baseData.message = "AI (Internal): 99 message:\"" + (message + " (" + targetSrc + ")").replace(/\"/g, strEmpty) + "\"";
+            baseData.properties = {
+                endpoint: endpointUrl
+            };
+            
+            return envelope;
+        }
+
+        function _createException(iKey:string, message:string, targetSrc:string, endpointUrl:any) {
+            let envelope : IEnvelope = _createEnvelope(iKey, "Exception");
+            let data = envelope.data;
+            data.baseType = "ExceptionData";
+            data.baseData.exceptions = [{
+                typeName: "SDKLoadFailed",
+                message: message.replace(/\./g, "-"),  // Replacing '.' characters as it causes the portal to hide the start of the message in the summary
+                hasFullStack: false,
+                stack: message + "\nSnippet failed to load [" + targetSrc + "] -- Telemetry is disabled\nHelp Link: " + helpLink + "\nHost: " + (locn && locn.pathname || "_unknown_") + "\nEndpoint: " + endpointUrl,
+                parsedStack: []
+            }];
+
+            return envelope;
+        }
+
+        let domainRetryIndex = -1;
+        let domainRetryCount = 0;
+        let domains = [
+            "js.monitor.azure.com",
+            "js.cdn.applicationinsights.io",
+            "js.cdn.monitor.azure.com",
+            "js0.cdn.applicationinsights.io",
+            "js0.cdn.monitor.azure.com",
+            "js2.cdn.applicationinsights.io",
+            "js2.cdn.monitor.azure.com",
+            "az416426.vo.msecnd.net" // this domain is supported but not recommended
+        ];
+
+        fetch("https://js.monitor.azure.com/beta/ai.3.integrity.json")
+            .then(response => response.json())
+            .then(json => {
+                let currentVersion = json.version;
```

Comment:
Rather than use the "version" it should use the value from the `ext.@<type>.file` and we will need a new config for the `<type>`  which will be one of the values in the file `@cjs.js`, `@cjs.min.js`, `@gbj.js`, `@gbl.min.js`, `@js` or `@min.js`
It will also need to grab the `integrity` value and include that as part of the `script` tag, as we are providing 2 assertions with this change
1) We will load the "full" version of the file for better page load (because it will be cached)
2) We will be adding the "integrity" to allow the browser to validate the script for better security.

For the second option, we will also need a configuration to allow the user to opt-out of adding the integrity

I've removed all of the hashes from the below copy of the current integrity file.

```json
{
    "name": "ai",
    "version": "3.2.0",
    "ext": {
        "@cjs.js": {
            "file": "ai.3.2.0.cjs.js",
            "type": "text/javascript; charset=utf-8",
            "integrity": "sha256-...==",
...
        },
        "@cjs.min.js": {
            "file": "ai.3.2.0.cjs.min.js",
            "type": "text/javascript; charset=utf-8",
            "integrity": "sha256-...==",
...
        },
        "@gbl.js": {
            "file": "ai.3.2.0.gbl.js",
            "type": "text/javascript; charset=utf-8",
            "integrity": "sha256-...==",
...
        },
        "@gbl.min.js": {
            "file": "ai.3.2.0.gbl.min.js",
            "type": "text/javascript; charset=utf-8",
            "integrity": "sha256-...==",
...
        },
        "@js": {
            "file": "ai.3.2.0.js",
            "type": "text/javascript; charset=utf-8",
            "integrity": "sha256-...==",
...
        },
        "@min.js": {
            "file": "ai.3.2.0.min.js",
            "type": "text/javascript; charset=utf-8",
            "integrity": "sha256-...==",
...
        }
    }
}